[Devtools Week] Update contributing guide (#39)
devfile/alizer: CONTRIBUTING.md

You can tag Alizer related issues with the `/area alizer` text in your issue.

### Development

#### Repository Format

The `alizer` repository includes different components:

- [CLI](./README.md#cli)
- [Alizer Library](./README.md#library-package)

As a result `alizer` can be used both as a cli tool and imported as a package inside other projects. More information for the repository can be found [here](./docs/public/alizer-spec.md)

#### Building locally

More information for building & running locally the project can be found [here](./README.md#usage)

#### Testing

Apart from testing your changes locally with an updated Alizer CLI, someone can test their changes by running `make test`. This will test the updates against all existing test cases.

### Submitting Pull Request

**Note:** All commits must be signed off with the footer:

```
Signed-off-by: First Lastname <email@email.com>
```

You can easily add this footer to your commits by adding `-s` when running `git commit`. When you think the code is ready for review, create a pull request and link the issue associated with it.

> Owners of the repository will watch out for and review new PRs.

Reviewer: ```suggestion
Owners of the repository will watch out for new PRs and provide reviews to them.
```

> By default for each change in the PR, GitHub Actions and OpenShift CI will run checks against your changes (linting, unit testing, and integration tests).

Author: @michael-valdron I've removed the openshift CI as is doesn't exist as a check and added the code coverage checks.

If comments have been given in a review, they have to be addressed before merging.

> After addressing review comments, don't forget to add a comment in the PR afterward, so everyone gets notified by Github and know to re-review.

Reviewer: @thepetk I'm thinking the author should mention the reviewer in their comment so the reviewer can provide a follow up review, wdyt?

```suggestion
After addressing review comments, don't forget to add a comment in the PR with the reviewer mentioned afterward, so they get notified by Github to provide a re-review.
```

# Contact us

> If you have questions, please visit us on `#devfile` in the [Kubernetes Slack](https://slack.k8s.io).

Reviewer: ```suggestion
If you have any questions, please visit us the `#devfile` channel under the [Kubernetes Slack](https://slack.k8s.io) workspace.
```
